### PR TITLE
Update old docs links

### PR DESCRIPTION
**Pull request by neverett**
# TL;DR

This PR updates outdated docs links, since those links will break once we switch this repo (and others) to the new monodocs build process.

## Type
- [ ] Bug Fix
- [ ] Feature
- [ ] Plugin

## Are all requirements met?

- [ ] Code completed
- [ ] Smoke tested
- [ ] Unit tests added
- [ ] Code documentation added
- [ ] Any pending items have an associated Issue

## Complete description
_How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Review by pingsutw — APPROVED** [review]
Thanks! @neverett could you also help update the old doc links in flytekit?
https://github.com/flyteorg/flytekit/blob/master/README.md

**Comment by neverett** [comment]
@pingsutw yep, there's a PR for that already: https://github.com/flyteorg/flytekit/pull/2210
